example/ui_showcase.py

```diff
@@ -149,11 +149,17 @@ async def do_magic(_v, _o) -> None:
 
 #############################################################################################
 # Overview page                                                                             #
-# Here we show the ImageMap and HideRows                                                    #
+# Here we show the ImageMap, HideRows and sub menus.                                        #
 #############################################################################################
 
 overview_page = web_server.page('overview', "Overview", menu_entry='Some Submenu', menu_icon='tachometer alternate',
-                                menu_sub_label="Overview")
+                                menu_sub_label="Overview", menu_sub_icon='tachometer alternate')
+
+submenu_page2 = web_server.page('submenu2', "Nothing here", menu_entry='Some Submenu', menu_sub_icon='couch',
+                                menu_sub_label="Empty Submenu")
+
+submenu_page3 = web_server.page('submenu3', "Nothing here either", menu_entry='Some Submenu',
+                                menu_sub_icon='motorcycle', menu_sub_label="Empty Submenu 2")
```

Comment:
You can also create menu entries without creating actual pages, using [`add_menu_entry()`](https://smarthomeconnect.readthedocs.io/en/latest/web.html#shc.web.interface.WebServer.add_menu_entry).

More specifically, I would suggest to add two links to the same page here for checking the correct behaviour in this case.

Reply:
Done.

Actually the WebServer API `page(...)` and `add_menu_entry(...)` is kind of confusing and not really obvious, e.g. that in `add_menu_entry()` the existing menu_entry is found by the label.  Maybe we should align the attribute names in both methods.

| page()| add_menu_entry()|
|-----------------|-----------------|
| menu_entry       | label|
| menu_icon| icon|
| menu_sub_label| sub_label|
| menu_sub_icon| sub_icon|

Not sure whether this would have avoided some of my confusion but we should fix it anyhow.  Maybe not now since this is an API change or what do you think?

Reply:
Hm, I'm actually quite happy with the current naming of these attributes: The `page()` method also takes `menu_entry=True` to simply create a menu entry with the page's title as label. This is why the parameter is called this way.

Basically, the `page()` method only creates a WebPage and the `add_menu_entry()` only adds a menu entry. Both are completely independent from each other. As a shortcut, the `page()` method allows to create a matching menu entry for the new WebPage in addition to the WebPage itself, by using the `menu_entry` parameter.

Maybe, we should point out this relationship in the docstrings of both methods, with a reference to the other.

Reply:
> More specifically, I would suggest to add two links to the same page here for checking the correct behaviour in this case.

Well this brings up another issue that we should clarify.  Since I identify the current page by the path in that case both menu entries are marked:

<img width="325" alt="image" src="https://github.com/user-attachments/assets/a59055eb-edac-488a-b0ef-23ed5a02884f">

This could be a feature or a bug depending on the point of view.  From my perspective I would say that is something that could be benefical and that I could live with.  Especially if the same page is referenced in different submenus this might be an interesting information being visualized like that.

Reply:
> Hm, I'm actually quite happy with the current naming of these attributes: The `page()` method also takes `menu_entry=True` to simply create a menu entry with the page's title as label. This is why the parameter is called this way.
> 
> Basically, the `page()` method only creates a WebPage and the `add_menu_entry()` only adds a menu entry. Both are completely independent from each other. As a shortcut, the `page()` method allows to create a matching menu entry for the new WebPage in addition to the WebPage itself, by using the `menu_entry` parameter.
> 
> Maybe, we should point out this relationship in the docstrings of both methods, with a reference to the other.

Done.